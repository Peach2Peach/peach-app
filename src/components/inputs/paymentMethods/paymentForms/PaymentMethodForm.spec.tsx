import { PaymentMethodForm } from '.'
import { createRenderer } from 'react-test-renderer/shallow'
import React from 'react'
import { View } from 'react-native'

jest.mock('react-native-gradients', () => ({
  LinearGradient: 'LinearGradient',
}))
jest.mock('../../../animation', () => ({
  Fade: ({ children }: any) => <View>{children}</View>,
}))

describe('PaymentMethodForm', () => {
  const renderer = createRenderer()
  const saveForm = jest.fn()
  jest.spyOn(React, 'useRef').mockReturnValue({ current: { save: saveForm } })
  it('should render correctly', () => {
    renderer.render(<PaymentMethodForm paymentMethod="sepa" data={{}} onSubmit={jest.fn()} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
