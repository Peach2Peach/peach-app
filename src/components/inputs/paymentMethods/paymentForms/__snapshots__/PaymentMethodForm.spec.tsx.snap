// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethodForm should render correctly 1`] = `
<View
  style={
    [
      {
        "height": "100%",
      },
      undefined,
    ]
  }
>
  <PeachScrollView
    contentContainerStyle={
      [
        {
          "flexGrow": 1,
          "justifyContent": "center",
        },
        {
          "paddingBottom": 40,
          "paddingTop": 16,
        },
      ]
    }
  >
    <Template1
      currencies={[]}
      data={{}}
      onSubmit={[Function]}
      paymentMethod="sepa"
      setFormData={[Function]}
      setStepValid={[Function]}
    />
  </PeachScrollView>
  <Fade
    show={true}
    style={
      {
        "alignItems": "center",
        "marginBottom": 40,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "height": 40,
          "marginTop": -40,
          "width": "100%",
        }
      }
    >
      <LinearGradient
        angle={90}
        colorList={
          [
            {
              "color": "#FFF9F6",
              "offset": "0%",
              "opacity": "1",
            },
            {
              "color": "#FFF9F6",
              "offset": "100%",
              "opacity": "0",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexGrow": 1,
        }
      }
    >
      <PrimaryButton
        disabled={true}
        narrow={true}
        onPress={[Function]}
        testID="navigation-next"
      >
        confirm
      </PrimaryButton>
    </View>
  </Fade>
</View>
`;
